packages/blade/src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,221 @@
+ *
+ * {showTag && (
+ *  <Tag
+ *    icon={CheckIcon}
+ *    onDismiss={() => setShowTag(false)}
+ *  >
+ *    Transactions
+ *  </Tag>
+ * )}
+ * ```
+ *
+ * Checkout [Tags Documentation](https://blade.razorpay.com/?path=/story/components-tag--default) for more info.
+ *
+ */
+const Tag = ({
+  size = 'medium',
+  icon: Icon,
+  onDismiss,
+  children,
+  isDisabled,
+  testID,
+  _isTagFocussed,
+  ...styledProps
+}: TagProps): React.ReactElement | null => {
+  const textColor = isDisabled
+    ? 'surface.text.placeholder.lowContrast'
+    : 'surface.text.subtle.lowContrast';
+
+  const mediumPadding: PaddingProps['padding'] = {
+    base: ['spacing.2', 'spacing.3', 'spacing.2', 'spacing.4'],
+    m: ['spacing.1', 'spacing.2', 'spacing.1', 'spacing.3'],
+  };
+
+  const largePadding: PaddingProps['padding'] = {
+    base: ['spacing.2', 'spacing.3', 'spacing.2', 'spacing.4'],
+    m: ['spacing.2', 'spacing.3', 'spacing.2', 'spacing.4'],
+  };
+
+  const getLeadingIcon = ({ size }: { size: IconProps['size'] }): React.ReactElement | null =>
+    Icon ? (
+      <Box display="flex" flexDirection="row" alignItems="center">
+        <Icon color={textColor} size={size} marginRight="spacing.2" />
+      </Box>
+    ) : null;
+
+  const getTagText = ({
+    size,
+  }: {
+    size: TextProps<{ variant: 'body' }>['size'];
+  }): React.ReactElement => (
+    <Box
+      display="flex"
+      flexDirection="row"
+      alignItems="center"
+      maxWidth={makeSize(globalSizeTokens['100'])}
+    >
+      <Text truncateAfterLines={1} marginRight="spacing.2" color={textColor} size={size}>
+        {children}
+      </Text>
+    </Box>
+  );
+
+  const getCloseIcon = ({ size }: { size: IconButtonProps['size'] }): React.ReactElement => (
+    <Box display="flex" flexDirection="row" alignItems="center" justifyContent="center">
+      <IconButton
+        size={size}
+        icon={CloseIcon}
+        accessibilityLabel={`Close ${children} tag`}
+        isDisabled={isDisabled}
+        onClick={() => {
+          onDismiss();
+        }}
+      />
+    </Box>
+  );
+
+  return (
+    <BaseBox
+      display={(isReactNative() ? 'flex' : 'inline-flex') as never}
+      {...getStyledProps(styledProps)}
+      {...metaAttribute({ name: MetaConstants.Tag, testID })}
+    >
+      <FocussableTag
+        display={(isReactNative() ? 'flex' : 'inline-flex') as never}
+        alignSelf={isReactNative() ? 'center' : undefined}
+        flexDirection="row"
+        flexWrap="nowrap"
+        backgroundColor="brand.gray.a100.lowContrast"
+        borderRadius="max"
+        padding={size === 'medium' ? mediumPadding : largePadding}
+        isTagFocussed={_isTagFocussed}
+      >
+        <ShowOnDesktop>
+          {getLeadingIcon({ size: 'small' })}
+          {getTagText({ size: 'small' })}
+          {getCloseIcon({ size: 'small' })}
+        </ShowOnDesktop>
+        <ShowOnMobile>
+          {getLeadingIcon({ size: size === 'large' ? 'medium' : 'small' })}
```

Comment:
why can't we just compute size based on screen and then pass that directly instead of 2 separate components?

Reply:
That will make size dependent on client-side javascript. I wanted to avoid that since it changes height and can cause some layout shift in SSR apps. This one uses media query to switch sizes.

Reply:
they layout shift will anyways get triggered because our typography scale is diff for mobile and desktop and that gets evaluated client side only 🤔

Reply:
> they layout shift will anyways get triggered because our typography scale is diff for mobile and desktop and that gets evaluated client side only 🤔

Lollll didn't know about this. Yup then just isMobile check makes more sense.

Reply:
updated the code to use isMobile